marm-mcp-server/marm_mcp_server/core/memory.py

```
@@ -367,18 +443,82 @@ async def auto_classify_content(self, content: str) -> str:
+                "content_preview": new_content[:100],
+            })
+
+            merged_hash = compute_content_hash(merged_content)
+
+            # Recompute embedding — keep existing if encoder unavailable
+            merged_embedding_bytes = None
+            encoder_ok = merged_content.strip() and self._load_encoder_lazily()
+            if encoder_ok:
+                try:
+                    merged_embedding_bytes = self.encoder.encode(merged_content).tobytes()
+                except Exception as e:
```

Ruff (0.15.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)